File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.java
class Solution {
    public int search(int[] nums, int target) {
        int FindDivisionIndex = 0 ; 
        for(int i = nums.length - 1 ; i >= 1 ; i--)
        {
            if(nums[i] < nums[i-1]) {FindDivisionIndex = i ; break;}
        }
        int index = FindDivisionIndex;
        for(int i = 0 ; i <= index; )
        {
            int middle = (i+index)/2 ; 
            if(nums[middle] == target) return middle ; 
            else if (nums[middle] < target) i = middle + 1 ; 
            else index = middle - 1 ; 
        }

        int index2 = FindDivisionIndex ; 
        for(int i = nums.length-1 ; i >= index2 ;)
        {
            int middle = (i+index2)/2 ; 
            if(nums[middle] == target) return middle ; 
            else if (nums[middle] < target) index2 = middle + 1 ; 
            else i = middle - 1 ; 
        }

        return -1 ; 
        
    }
}

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-1.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
         
        def QuickSort(nums:List[int])->List[int]:
            if(len(nums) <= 1):return nums
            value : int = nums[len(nums)//2]
            return QuickSort([num for num in nums if num > value])+[num for num in nums if num == value]+QuickSort([num for num in nums if num < value])

        return QuickSort(nums)[k-1]


        


                    




     


        
        